### PR TITLE
Resync midstream serverless-1.23 branch

### PR DESCRIPTION
**Pull request by dsimansk**
- Revert initial OpenShift files commit
- Redo OpenShift checkout + patch apply
- Regen templates.

For CI runs.

/hold

### CODE REVIEW

File: templates/go/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Optional list of additional Buildpacks to be applied to the language pack's
+# builder image when the Function is built using a Buildpack builder.
+buildpacks:
+  - paketo-buildpacks/go-dist
+  - ghcr.io/boson-project/go-function-buildpack:tip
```
Comment (dsimansk) [inline templates/go/manifest.yaml:5]: @lance just to be sure, is this OK? It seems redundant with `DefaultBuilderImages` map above.

Comment (lance) [reply templates/go/manifest.yaml:5]: Yes - this is the same as in the upstream: https://github.com/knative-sandbox/kn-plugin-func/blob/main/templates/go/manifest.yaml. It is additive to what is in the `DefaultBuilderImages` map. Those are builder images, and these are buildpacks that are applied to the builder image at build time.

File: templates/rust/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+buildpacks:
+  - docker.io/paketocommunity/rust
```
[the remark this answers is not in the document]
Comment (lance) [reply templates/rust/manifest.yaml:2]: Same as above. This is a buildpack, the map contains builders.

**Comment by lance** [comment]
@dsimansk I think as part of this pull request you should also remove this entire directory https://github.com/openshift-knative/kn-plugin-func/tree/main/templates/quarkus

And be sure to regenerate the `zz_*.go` file please. 🙏

**Comment by lance** [comment]
> @lance I've got the templates from upstream release. I'll create a separate PR to `main` to remove the override.

Oh right - that's what's happening here. ✅ My mistake, carry on. 😄

**Review by lance — COMMENTED** [review]
/lgtm

**Pull request merged**